### DISCUSSION TITLE
Feature request: Modify Events by BT, not time

### DISCUSSION THREAD

**Discussion by fatrabbit-la** [artisan-roaster-scope/artisan#985] [Ideas]
Whenever I need to add or modify events, I do them relative to BT but it appears I have to add/modify by time. Example: I wanted to add an event that occurred at 300F but rather than adjusting it by BT, I had to guess what time BT hit that temp and continue adjusting the time until it lined up with the BT I wanted.

Thanks for considering this option.

**Comment by poundy** [comment]
For me I always just use the mouse to get my cursor in the right / desired spot and then right click to add the event. From the mouse display I can see the time and temp that I am adding the value to. 
I understand your request - but I suspect that the data would likely need to be transformed to do this, as time is the key dimension... what might be possible is to have the program interpolate data points either side of that event and choose the approximate time that corresponded to when your data says that temp should have occurred, so you don't have to do the adjustments manually

**Comment by roasterdave** [comment]
The procedure described above by poundy is quick and easy and works well for many situations.

You might want to correlate your event to just about anything, not just BT. The cursor widget and the mouse cross hairs can be used to find the time for anything in the graph. Use that time value to update your event.

The cursor widget was enhanced in v2.6.0 and is described in the release **[blog post](https://artisan-roasterscope.blogspot.com/2022/03/artisan-v260.html)**. Scroll down to the Cursor Widget section.

Both the cursor widget and the mouse cross hairs are described in detail in the video **[Navigating Artisan - Getting the Most From Your Artisan Curves](https://www.youtube.com/watch?v=tJKRJrEPeEQ)**. Click '...more' in the description to expand it to see the chapter index. This link will take you directly to the **[Mouse Cursor](https://www.youtube.com/watch?v=tJKRJrEPeEQ&t=785s)** section of the video.

**Comment by fatrabbit-la** [comment]
Thanks for that tip. I think being able to do it in the Events tab by BT would be a bit more efficient but this is a much better solution than the way I was doing it.

Follow up question…what does the ‘Cluster’ button do?

Dan 

Fat Rabbit Coffee
***@***.*** ***@***.***>
www.fatrabbitcoffee.com <http://www.fatrabbitcoffee.com/>

**Reply by roasterdave** [reply]
The Cluster feature is useful if you use event quantifiers to convert readings from a connected device to display as one of the four special events (**[quick start guide](https://artisan-scope.org/docs/events/), [blog](https://artisan-roasterscope.blogspot.com/2014/04/event-quantifiers.html)**).  With some devices this can result in redundant event entries.  The Cluster button on the Roast>> Properties>> Events tab eliminates these entries after the roast by clustering them into one entry.  There is also a Cluster check box on the Config>> Events>> Quantifiers tab to perform this action during the roast.  For more information see this **[blog post](https://artisan-roasterscope.blogspot.com/2017/02/artisan-v10.html)**.

**Comment by MAKOMO** [comment]
As it is unclear what point in time editing by temperature should indicate if a temperature occurs several time in the profile I reject this feature request for now.
